operator/v1/composite_operator.go

```diff
@@ -184,6 +184,7 @@ func (co *CompositeOperator) Cleanup(ctx context.Context, obj client.Object) (re
 	defer span.End()
 
 	if !co.IsSuspended(ctx, obj) {
+		defer co.order.Reverse()
```

Comment:
The bug seems to be in the `Reverse()` method itself. It should return a reverse ordered version of the OperandOrder, and not reverse the main OperandOrder itself.

Can you change the implementation in https://github.com/darkowlzz/operator-toolkit/blob/aeb71b541f96d315b816c859d7bdafb37605aceb/operator/v1/operand/order.go#L41-L48 to something like:

```go
      x := make([][]Operand, len(o))
      // Refer: https://github.com/golang/go/wiki/SliceTricks#reversing
      for left, right := 0, len(o)-1; left < right; left, right = left+1, right-1 {
            x[left], x[right] = o[right], o[left]
      }
      return x
```

Tried this change with dag_test.go
```go
	fmt.Println("ORDER:", ordered.String())
	reverseOrder := ordered.Reverse()
	fmt.Println("REVERSE:", reverseOrder.String())
	fmt.Println("ORDER:", ordered.String())
```
and got:

```console
ORDER: [
  0: [ A B ]
  1: [ C ]
  2: [ D F ]
  3: [ E ]
]
REVERSE: [
  0: [ E ]
  1: [ D F ]
  2: [ C ]
  3: [ A B ]
]
ORDER: [
  0: [ A B ]
  1: [ C ]
  2: [ D F ]
  3: [ E ]
]
```

Reply:
Would be great if the `TestDAG()` is updated to check this. After the last reverse, check to ensure the main order is still the same.